Kitodo-API/src/main/java/org/kitodo/api/dataformat/mets/InputStreamProviderInterface.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * (c) Kitodo. Key to digital objects e. V. <contact@kitodo.org>
+ *
+ * This file is part of the Kitodo project.
+ *
+ * It is licensed under GNU General Public License version 3 or later.
+ *
+ * For the full copyright and license information, please read the
+ * GPL3-License.txt file that was distributed with this source code.
+ */
+
+package org.kitodo.api.dataformat.mets;
+
+import java.io.InputStream;
+import java.net.URI;
+
+/**
+ * A functional interface for a function
+ * {@code InputStream getInputStream(URI uri, boolean couldHaveToBeWrittenInTheFuture)}.
+ */
+public interface InputStreamProviderInterface {
+
+    /**
+     * A function that opens an input stream from a source identified by an URI.
+     * If used, the calling function is responsible of closing the stream.
+     *
+     * @param uri
+     *            the identifier for the file
+     * @param couldHaveToBeWrittenInTheFuture
```

Comment:
Sorry, i don't get this one.
could have to be written in the future?
"Es könnte auch in der Zukunft geschrieben worden sein?"

Did you mean
"Needs to be written in the future"?

Reply:
„Könnte in der Zukunft geschrieben werden müssen.“ If so, the lock must be requested so that no other metadata editor edits the file at the same time. If not, a read only copy can also be used. Then another process can change the file at the same time, the current editor would only see this after a reload.